### PR TITLE
Implement pagination

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Implement a custom hook of usePagination and use it on the publication page (layout by category and all publication), import publication page and scholly 

# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
The current publication page has no pagination implemented, too many publications might have an issue of scrolling a lot

# Solution description
See description

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any) 
N/A

# Risk
Low for publication page. High for import publication page as lots of functions had been removed.

### CODE REVIEW

**Review by kdav108 — APPROVED** [review]
Nice job with the custom hook, the codebase looks good and it works fine for me!

File: client/src/components/publications/publicationsLayout/LayoutByCategory.js [hunk 1]
```diff
@@ -14,11 +15,10 @@ const LayoutByCategory = ({ teamPublications }) => {
...
           <h2 className="publicationListHeader"> {categoryType} </h2>
-          <div className="publicationList">
-            {publicationsByCategory.map((pub) => (
-              <Publication pub={pub} key={pub._id} />
-            ))}
-          </div>
+          <LayoutAllPublications 
+            teamPublications={publicationsByCategory} 
+            pageSize={categoryPageSize}
+          />;
```
Comment (vgoh1) [inline client/src/components/publications/publicationsLayout/LayoutByCategory.js:21]: random semicolon here

**Review by vgoh1 — APPROVED** [review]
Thanks for cleaning up my roundabout pagination implementation! The import publications and normal publications pagination works fine for me.

File: client/src/components/publications/publicationsLayout/LayoutByCategory.js (hunk 1 above)
Comment (pmyip3) [reply client/src/components/publications/publicationsLayout/LayoutByCategory.js:21]: Fixed. Thanks for catching it!

**Review by jfan0010 — APPROVED** [review]

**Pull request merged**